src/lib/useEffectOnce.ts

```diff
@@ -0,0 +1,14 @@
+import { useEffect, useRef } from "react";
+
+function useEffectOnce(effect) {
+  const hasRunRef = useRef(false);
+
+  useEffect(() => {
+    if (!hasRunRef.current) {
+      effect();
+      hasRunRef.current = true;
+    }
+  }, [effect]);
+}
+
+export default useEffectOnce;
```

Comment:
this hook seems like a workaround, are you sure we need to re-use it somewhere else?

Reply:
The problem forced me to create this helper was that i'm unable to set an initial state on a component mount when i use a centralized state & selectors. I agree that it's not a best solution. Ideally, we should maintain state independently of whether the component is mounted or not. But at the moment it makes code a bit more readable. I'll have no problem removing it when it will be no longer needed